source/Color.cpp

```diff
@@ -80,3 +80,12 @@ Color Color::Additive(float alpha) const
 	
 	return result;
 }
+
+Color Color::Combine(double a1, Color c1, double a2, Color c2)
```

Comment:
> (and arithmetic operators should be spaced, e.g. `(1 - percent) * a.color[0] + percent * b.color[0];`)

Fixed.



> Fractions sum to 1:
> 
> If we must keep both components, then we should ensure we clamp the results into a valid range.

Well, they don't _necessarily_ sum up to 1. If someone wants to write `0.3 * color1 + 0.5 * color2`, the API might as well let them. So how about, instead of clamping, ensure that the coefficients are non-negative and their sum is <= 1?

Or we could just change `Combine` to `Interpolate` as you suggest. I don't mind one way or the other.

Should I also declare the coefficients as floats while I'm at it?

Reply:
Yes, making them floats would be good.

I think in general there will not be much use of this method, as designing a UI where colors are blended in a specific hardcoded manner means that the result may not be compatible with theming plugins. (I do recognize that I basically asked for you to implement this method / use existing transforms.)

When we redo MapSalesPanel to derive all UI information from interface.txt definitions, I think this method will be removed, in favor of a more explicit visual representation of "stored" vs "for sale" vs "both", such as a specific sprite icon.

Reply:
Changed to floats. I understand this method may be obsoleted soon; that's how it is sometimes, no problem.